stake-pool/cli/src/client.rs

```diff
@@ -84,13 +84,12 @@ pub(crate) fn get_stake_pools(
     rpc_client
         .get_program_accounts_with_config(
             &spl_stake_pool::id(),
-            #[allow(deprecated)] // TODO figure out what encoding to use
             RpcProgramAccountsConfig {
-                filters: Some(vec![RpcFilterType::Memcmp(Memcmp {
-                    offset: 0, // 0 is the account type
-                    bytes: MemcmpEncodedBytes::Base58("2".to_string()),
-                    encoding: None,
-                })]),
+                // 0 is the account type
+                filters: Some(vec![RpcFilterType::Memcmp(Memcmp::new_raw_bytes(
+                    0,
+                    vec![1],
```

Comment:
i couldnt find tests for this program but i confirmed externally this should be right

```
use solana_program::pubkey::Pubkey;
use solana_client::rpc_filter::{Memcmp, MemcmpEncoding, MemcmpEncodedBytes};

fn main() {
    let key = Pubkey::new_rand();

    let first_t = Memcmp {
        offset: 0,
        bytes: MemcmpEncodedBytes::Base58("2".to_string()),
        encoding: None,
    };

    let first_o = Memcmp::new_raw_bytes(0, vec![1]);

    let second_t = Memcmp {
        offset: 12,
        bytes: MemcmpEncodedBytes::Base58(key.to_string()),
        encoding: Some(MemcmpEncoding::Binary),
    };

    let second_o = Memcmp::new_base58_encoded(12, &key.to_bytes());

    println!("first (theirs):  {:?}", first_t.bytes());
    println!("first (ours):    {:?}", first_o.bytes());
    println!("\nsecond (theirs): {:?}", second_t.bytes());
    println!("second (ours):   {:?}", second_o.bytes());
}
```

```
    Finished dev [unoptimized + debuginfo] target(s) in 13.13s
     Running `target/debug/memcmp-test`
first (theirs):  Some([1])
first (ours):    Some([1])

second (theirs): Some([172, 202, 17, 192, 239, 120, 241, 109, 190, 67, 90, 45, 39, 45, 102, 1, 126, 172, 5, 167, 168, 222, 203, 72, 209, 138, 25, 102, 201, 25, 217, 23])
second (ours):   Some([172, 202, 17, 192, 239, 120, 241, 109, 190, 67, 90, 45, 39, 45, 102, 1, 126, 172, 5, 167, 168, 222, 203, 72, 209, 138, 25, 102, 201, 25, 217, 23])
```